tests/test_dim_reduction.py

```diff
@@ -0,0 +1,100 @@
+import unittest 
+import pandas as pd 
+import numpy as np 
+import os 
+import sys 
+current_path = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(current_path + "/../neasqc_wp61/models/quantum/")
+from dim_reduction import *
+
+class TestDimReduction(unittest.TestCase):
+    
+    def setUp(self):
+        """
+        Generate random dataset for our tests
+        """
+        n_rows = 10 
+        data = {
+            'label' : np.random.randint(2, size=n_rows),
+            'sentence_embedding' : [
+                np.random.rand(768) for _ in range(n_rows)]
+        }
+        self.df = pd.DataFrame(data)
+
+
+    def testPCA(self):
+        """
+        Test of PCA dimensionality reduction. 
+        """
+        dim_out = 4
```

Comment:
The output dimension should NOT be hard-coded, it should be a parameter that you pass.
Also, make sure to test with a number of different parameters, there are ways to do that. You give a list of dimensions as input to your test, and it will be tested for each of those inputs.
We don't want out function to particularly always output an embedding of dim 4, we want it to output a dim N when we ask for a dim N reduction. Whatever that N may be.
 This is what your test should reflect.